lib/kennel/models/monitor.rb

```diff
@@ -108,6 +108,16 @@ def as_json
         @as_json = data
       end
 
+      def resolve_linked_tracking_ids(id_map)
+        if as_json[:type] == "composite"
+          begin
+            as_json[:query].gsub!(/%\{(.*?)\}/) { id_map.fetch($1) }
```

Comment:
safer to re-assign with gsub since the query that was passed in might be stored somewhere / be reused